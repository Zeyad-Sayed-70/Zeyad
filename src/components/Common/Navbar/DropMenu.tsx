"use client";
import React from "react";
import { VscMenu } from "react-icons/vsc";
import {
  DropdownMenu,
  DropdownMenuContent,
  DropdownMenuItem,
  DropdownMenuLabel,
  DropdownMenuSeparator,
  DropdownMenuTrigger,
} from "@/components/ui/dropdown-menu";
import { links_data } from "@/constants/navbar";
import { useRouter } from "next/navigation";

function DropMenu() {
  const router = useRouter();
  function handleClick(href: string) {
    router.push(href);
  }
  return (
    <>
      <DropdownMenu>
        <DropdownMenuTrigger>
          <VscMenu size={30} />
        </DropdownMenuTrigger>
        <DropdownMenuContent className="w-[250px]">
          {links_data.map((link) => (
            <DropdownMenuItem
              key={link.id}
              onSelect={(e: any) => handleClick(link.href)}
              className="text-lg"
            >
              {link.title}
            </DropdownMenuItem>
          ))}
          <DropdownMenuSeparator />
          <DropdownMenuLabel className="text-lg">
            Call To Action
          </DropdownMenuLabel>
          <DropdownMenuItem className="text-lg">Get In Touch</DropdownMenuItem>
        </DropdownMenuContent>
      </DropdownMenu>
    </>
  );
}

export default DropMenu;
