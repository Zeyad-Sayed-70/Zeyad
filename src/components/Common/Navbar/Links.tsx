import { links_data } from "@/constants/navbar";
import Link from "next/link";
import React from "react";

function Links() {
  return (
    <ul className="flex items-center gap-4">
      {links_data.map((link) => (
        <li
          key={link.id}
          className="font-semibold hover:text-primary hover:underline"
        >
          <Link href={link.href}>{link.title}</Link>
        </li>
      ))}
    </ul>
  );
}

export default Links;
