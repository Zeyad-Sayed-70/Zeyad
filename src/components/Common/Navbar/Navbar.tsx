"use client";
import { lazy, Suspense } from "react";
import Logo from "../Logo";
import Link from "next/link";
import { Button } from "../../ui/button";
import useMedia from "@/hooks/useMedia";
import Container from "../Container";

const Links = lazy(() => import("./Links"));
const SideMenu = lazy(() => import("./SideMenu"));

function Navbar() {
  const media = useMedia();

  return (
    <Container>
      <nav className="h-[70px] w-full flex items-center gap-12">
        <Logo width={100} />
        <Suspense>
          {/* {media && media >= 768 && ()} */}
          <>
            <div className="hidden md:block">
              <Links />
            </div>
            <div className="hidden md:block ml-auto">
              <Link href={"/contact"}>
                <Button
                  type="button"
                  role="contact"
                  className="text-lg py-5 bg-main hover:main-variant"
                >
                  Get In Touch
                </Button>
              </Link>
            </div>
          </>
        </Suspense>
        <Suspense>
          {/* {media && media < 768 && ()} */}
          <div className="block md:hidden ml-auto">
            <SideMenu />
          </div>
        </Suspense>
      </nav>
    </Container>
  );
}

export default Navbar;
