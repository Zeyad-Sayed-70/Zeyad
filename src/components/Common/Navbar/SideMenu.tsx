"use client";
import { useState } from "react";
import { Button } from "@/components/ui/button";
import { links_data } from "@/constants/navbar";
import { VscMenu, VscClose } from "react-icons/vsc";

import "./index.css";
import { useRouter } from "next/navigation";

function SideMenu() {
  const [isOpen, setIsOpen] = useState<boolean | null>(null);
  const router = useRouter();

  function handleClick(href: string) {
    router.push(href);
    setIsOpen(false);
  }

  return (
    <>
      <main>
        <Button
          type="button"
          role="open"
          aria-pressed={isOpen as boolean}
          variant="outline"
          className="text-2xl"
          onClick={() => setIsOpen(true)}
        >
          <VscMenu />
        </Button>
        <aside
          aria-expanded={isOpen as boolean}
          className={`w-full h-full fixed top-0 bg-slate-200 z-10 ${
            isOpen
              ? "slide-to-left"
              : isOpen === false
              ? "slide-to-right"
              : "-right-[120%]"
          }`}
        >
          <Button
            type="button"
            role="close"
            aria-pressed={isOpen as boolean}
            onClick={() => setIsOpen(false)}
            className="absolute top-4 right-4 text-xl bg-[none]"
            variant="outline"
          >
            <VscClose />
          </Button>

          <section className="h-full px-6 flex flex-col items-center justify-center gap-24">
            <ul className="w-full text-center flex flex-col gap-4">
              {links_data.map((link) => (
                <li
                  key={link.id}
                  className="w-full py-2 px-8 rounded-md cursor-pointer hover:bg-slate-300 text-4xl font-light"
                  onClick={() => handleClick(link.href)}
                >
                  {link.title}
                </li>
              ))}
            </ul>
            <Button
              type="button"
              role="contact"
              className="text-xl py-4"
              onClick={() => handleClick("/contact")}
            >
              Get In Touch
            </Button>
          </section>
        </aside>
      </main>
    </>
  );
}

export default SideMenu;
