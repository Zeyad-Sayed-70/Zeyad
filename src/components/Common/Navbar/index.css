.slide-to-left {
  animation: slide-to-left 0.5s both ease-in-out;
}

.slide-to-right {
  animation: slide-to-right 0.5s both ease-in-out;
}

@keyframes slide-to-left {
  from {
    right: -120%;
  }

  to {
    right: 0;
  }
}

@keyframes slide-to-right {
  from {
    right: 0;
  }

  to {
    right: -120%;
  }
}
