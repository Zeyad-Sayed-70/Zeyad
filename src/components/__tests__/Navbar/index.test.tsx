import Navbar from "@/components/Common/Navbar/Navbar";
import { render, screen } from "@testing-library/react";

test("render navbar component", () => {
  render(<Navbar />);

  expect(screen.getByRole("navigation")).toMatchSnapshot();
});
