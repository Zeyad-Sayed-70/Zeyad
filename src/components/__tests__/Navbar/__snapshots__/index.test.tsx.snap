// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render navbar component 1`] = `
<nav
  class="h-[70px] w-full flex items-center gap-12"
>
  <a
    href="/"
  >
    <img
      alt="ZEYAD Logo"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fassets%2Fzeyad_logo.png&w=256&q=75"
      srcset="/_next/image?url=%2Fassets%2Fzeyad_logo.png&w=128&q=75 1x, /_next/image?url=%2Fassets%2Fzeyad_logo.png&w=256&q=75 2x"
      style="color: transparent;"
      width="100"
    />
  </a>
</nav>
`;
